go/vt/vttablet/tabletmanager/rpc_replication.go

```diff
@@ -1123,6 +1142,15 @@ func (tm *TabletManager) handleRelayLogError(err error) error {
 // repairReplication tries to connect this server to whoever is
 // the current primary of the shard, and start replicating.
 func (tm *TabletManager) repairReplication(ctx context.Context) error {
+	if tm.isSetReplicationSourceLockedRunning() {
+		// we are actively setting replication source,
+		// repairReplication will block due to higher
+		// authority holding a shard lock (PRS on vtctld)
+		log.Infof("slack-debug: we are actively setting replication source, exiting")
+
+		return nil
```

Comment:
@vmogilev `return nil` makes the client unaware the repairReplication never happened. Should we return an error to the RPC caller that explains what happened?